lib/IRGen/GenStruct.cpp

```diff
@@ -823,8 +873,10 @@ class ClangRecordLowering {
   }
 
   /// Add information to track a value field at the current offset.
-  void addFieldInfo(VarDecl *swiftField, const LoadableTypeInfo &fieldType) {
-    unsigned explosionSize = fieldType.getExplosionSize();
+  void addFieldInfo(VarDecl *swiftField, const FixedTypeInfo &fieldType) {
+    unsigned explosionSize = SwiftDecl->getStoredProperties().size();
```

Comment:
If I understand it correctly, the concept of explosion does not make sense for address-only types, so it is not a surprise that we don't have an appropriate API to call here.

This information is eventually written into `RecordField::Begin` / `::End`. Note that only `ClangFieldInfo` is setting these fields. Even `StructFieldInfo` is not setting them. Since this data is not correct for native Swift structs, I suspect that users of this data are dead code. Could you take a look? It is quite possible you would be able to delete some/all of this stuff.

If this data is used, I think we should either change `ClangFieldInfo` to be like `StructFieldInfo` and not set the explosion begin/end, at least when we have an address-only type.

Reply:
Yeah, you're right! With a small amount of refactoring, I was able to get this working without `begin`/`end`.

Reply:
Turns out after running all the tests there are some things which rely on `begin`/`end`. At first, I was confused by this because, as you said, it doesn't appear that `StructFieldInfo` sets them. But, after further investigation, it appears they are set for `StructFieldInfo`. [Just in a more convoluted way.](https://github.com/apple/swift/blob/master/lib/IRGen/GenRecord.h#L868-L871)

I'll have to add back the old path, too.